modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -591,6 +591,21 @@ void ONNXImporter::populateNet(Net dstNet)
             }
             layerParams.set("num_output", layerParams.blobs[0].size[1] * layerParams.get<int>("group", 1));
             layerParams.set("bias_term", node_proto.input_size() == 3);
+
+            const DictValue* outShape = layerParams.ptr("output_shape");
+            if (outShape && outShape->size() == 4)
+            {
+                const int strideY = layerParams.get<int>("stride_h");
+                const int strideX = layerParams.get<int>("stride_w");
+                const int outH = outShape->getIntValue(2);
+                const int outW = outShape->getIntValue(3);
+
+                if (layerParams.get<String>("pad_mode") == "SAME")
+                {
+                    layerParams.set("adj_w", (outW - 1) % strideX);
+                    layerParams.set("adj_h", (outH - 1) % strideY);
+                }
```

Comment:
What about "VALID" mode? Please see https://github.com/opencv/opencv/blob/master/modules/dnn/src/tensorflow/tf_importer.cpp for a reference code.

Reply:
This is done too.